apps/web/utils/actions/ai-rule.ts

```
@@ -691,52 +697,54 @@ function shouldAutomate(actions: Pick<Action, "type">[]) {
-      },
-      { maxLength: 500 },
+    const gmail = getGmailClient(session);
+    const lastSent = await getMessages(gmail, {
+      query: "in:sent",
+      maxResults: 20,
+    });
+    const gmailLabels = await getLabels(gmail);
+    const userLabels = gmailLabels?.filter((label) => label.type === "user");
+    const lastSentMessages = await Promise.all(
+      lastSent.messages?.map(async (message) => {
+        const gmailMessage = await getMessage(message.id!, gmail);
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)